src/xgov-dapp/src/features/status/index.tsx

```diff
@@ -0,0 +1,444 @@
+                will be your payout in 12 months, if you remain eligible.
+              </Typography>
+            </div>
+          </Box>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+function VoteSessionRow({ globalState, termPools }: { globalState: VotingRoundGlobalState; termPools: TermPool[] }) {
+  const { activeAddress } = useWallet()
+
+  const [votingRoundMetadata, setVotingRoundMetadata] = useState<VotingRoundMetadata | undefined>(undefined)
+  const [voterVotes, setVoterVotes] = useState<string[] | undefined>(undefined)
+  const [isLoadingMetadata, setIsLoadingMetadata] = useState<boolean>(true)
+  const [isLoadingVotersVote, setIsLoadingVotersVote] = useState<boolean>(true)
+
+  const hasVoted = voterVotes !== undefined ? true : false
+  const hasVoteStarted = !globalState ? false : getHasVoteStarted(globalState)
+  const hasVoteEnded = !globalState ? false : getHasVoteEnded(globalState)
+  const canVote = hasVoteStarted && !hasVoteEnded
+
+  useEffect(() => {
+    ;(async () => {
+      setIsLoadingMetadata(true)
+      try {
+        setVotingRoundMetadata(await fetchVotingRoundMetadata(globalState.metadata_ipfs_cid))
+        setIsLoadingMetadata(false)
+      } catch (e) {
+        setIsLoadingMetadata(false)
+      }
+    })()
+  }, [globalState])
+
+  useEffect(() => {
+    refetchVotersVote(globalState.appId, activeAddress, votingRoundMetadata, globalState)
+  }, [globalState, activeAddress, votingRoundMetadata])
+
+  const refetchVotersVote = async (
+    voteId: number | undefined,
+    walletAddress: string | undefined,
+    votingRoundMetadata: VotingRoundMetadata | undefined,
+    votingRoundGlobalState: VotingRoundGlobalState | undefined,
+  ) => {
+    if (voteId && walletAddress && votingRoundMetadata && votingRoundGlobalState) {
+      setIsLoadingVotersVote(true)
+      try {
+        setVoterVotes(await fetchVoterVotes(voteId, walletAddress, votingRoundMetadata, votingRoundGlobalState))
+        setIsLoadingVotersVote(false)
+      } catch (e) {
+        setIsLoadingVotersVote(false)
+        // handleError(e)
+      }
+    } else {
+      setIsLoadingVotersVote(false)
+      setVoterVotes(undefined)
+    }
+  }
+
+  if (isLoadingMetadata || isLoadingVotersVote) {
+    return (
+      <div className="table-row">
+        <div className="table-cell">
+          <Skeleton className="h-14 w-full mt-2" variant="rectangular" />
+        </div>
+        <div className="table-cell">
+          <Skeleton className="h-14 w-full mt-2" variant="rectangular" />
+        </div>
+        <div className="table-cell">
+          <Skeleton className="h-14 w-full mt-2" variant="rectangular" />
+        </div>
+        <div className="table-cell">
+          <Skeleton className="h-14 w-full mt-2" variant="rectangular" />
+        </div>
+        <div className="table-cell">
+          <Skeleton className="h-14 w-full mt-2" variant="rectangular" />
+        </div>
```

Comment:
very repetitive - consider a component or at the very something like `range(4).map(() => (<my-repeating-markup>...)` or similar